src/roblox_api/instance.rs

```diff
@@ -99,17 +103,18 @@ impl LuaInstance {
         })?;
 
         let mut descendants = Vec::new();
-        let mut stack = vec![instance];
+        let mut stack = VecDeque::from_iter(instance.children().into_iter());
 
-        while let Some(current) = stack.pop() {
-            for &child_ref in current.children() {
-                let child_instance = tree
-                    .get_by_ref(child_ref)
-                    .expect("received invalid child in tree when recursing through descendants");
+        while let Some(current) = stack.pop_front() {
+            descendants.push(LuaInstance::new(Arc::clone(&self.tree), *current));
 
-                descendants.push(LuaInstance::new(Arc::clone(&self.tree), child_ref));
-                stack.push(child_instance);
-            }
+            let current_instance = tree
+                .get_by_ref(*current)
+                .expect("received invalid child in tree when recursing through descendants");
+
+            let mut new_stack = VecDeque::from_iter(current_instance.children());
+            new_stack.extend(&stack);
+            stack = new_stack;
```

Comment:
Can we use `push_front` here instead? Whenever possible, reusing the same container for an algorithm like this is much easier to reason about than allocating a new one and appending all other elements onto it.

Reply:
Not unless we go in reverse when iterating children, is that fine?